app/scripts/metamask-controller.js

```diff
@@ -29,38 +30,48 @@ module.exports = class MetamaskController extends EventEmitter {
   constructor (opts) {
     super()
     this.opts = opts
-    this.state = { network: 'loading' }
     let initState = opts.initState || {}
 
     // observable state store
     this.store = new ObservableStore(initState)
 
+    // network store
+    this.networkStore = new ObservableStore({ network: 'loading' })
+
     // config manager
     this.configManager = new ConfigManager({
       store: this.store,
     })
-    this.configManager.updateConversionRate()
 
     // preferences controller
     this.preferencesController = new PreferencesController({
       initState: initState.PreferencesController,
     })
 
+    // currency controller
+    this.currencyController = new CurrencyController({
+      initState: initState.CurrencyController,
+    })
+    this.currencyController.updateConversionRate()
+    this.currencyController.scheduleConversionInterval()
```

Comment:
Would be slightly cool if this accepted an interval argument, and if could be passed to the metamask-controller constructor, just for making the whole lib configurable. (non blocking, just a thought).